test/client_tests/client_test.dart

```diff
@@ -286,13 +289,86 @@ void main() {
     );
   });
 
+  test('Call throws if HTTP status indicates an error', () async {
+    void handleRequest(_) {
+      harness.toClient.add(HeadersStreamMessage([
+        Header.ascii(':status', HttpStatus.serviceUnavailable.toString()),
+        Header.ascii('content-type', 'application/grpc'),
+      ]));
+      // Send a frame that might be misinterpreted as header frame and cause
```

Comment:
I think it cannot be misinterpreted as header frame - http/2 will treat it as data frame(s).

as a header frame -> as a length-prefixed proto message?

(also below)